### ISSUE TITLE
spacewalk-common-channels error no such option refresh-channels

### ISSUE THREAD

**Issue by saltyollpheist** [uyuni-project/uyuni#5022]
Following documentation provided at [Registering Ubuntu 16.04 and 18.04 Clients](https://www.uyuni-project.org/uyuni-docs/en/uyuni/client-configuration/clients-ubuntu-old.html). Using the syntax provided under **Procedure: Adding Software Channels at the Command Prompt**, I entered the following into CLI: 

`spacewalk-common-channels \`
`ubuntu-1804-pool-amd64-uyuni \`
`ubuntu-1804-amd64-main-uyuni \`
`ubuntu-1804-amd64-main-updates-uyuni \`
`ubuntu-1804-amd64-main-security-uyuni \`
`ubuntu-1804-amd64-universe-uyuni \`
`ubuntu-1804-amd64-universe-updates-uyuni \`
`ubuntu-1804-amd64-universe-security-uyuni \`
`ubuntu-1804-amd64-uyuni-client \`
`mgr-sync refresh --refresh-channels`

Upon confirming the last line, I'm shown the following error message: 

`Usage: spacewalk-common-channels [options] <channel1 glob> [<channel2 glob> ... ]`
`spacewalk-common-channels: error: no such option: --refresh-channels`

I've successfully installed Uyuni on my home server and was able to follow these steps to the letter without any need to register anything on the SCC, and it was able to add these software channels without encountering this error. Same build, 2022.01. Using this build due to known Time Zone bug, both for home server and work lab. 

Uyuni 2022.01; No proxy used.

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-stable
Name           : Uyuni-Server-release
Version        : 2022.01-165.2.uyuni1
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2022.01-165.2.uyuni1.src
Summary        : Uyuni Server
Description    :

Performed `tail -f` on the following log files with no results when processing the above commands: 
mgr-sync.log
reposync.log
rhn_web_ui.log

**Comment by mackdk** [comment]
From what I see, you have entered the wrong command. The problem is the last backslash at the end of the line `ubuntu-1804-amd64-uyuni-client`, which wrongly combines two separate commands into one.

This procedure involves two steps:

1) The first step uses `spacewalk-common-channels` to add the channels:

  ```
  spacewalk-common-channels \
  ubuntu-1804-pool-amd64-uyuni \
  ubuntu-1804-amd64-main-uyuni \
  ubuntu-1804-amd64-main-updates-uyuni \
  ubuntu-1804-amd64-main-security-uyuni \
  ubuntu-1804-amd64-universe-uyuni \
  ubuntu-1804-amd64-universe-updates-uyuni \
  ubuntu-1804-amd64-universe-security-uyuni \
  ubuntu-1804-amd64-uyuni-client
  ```

2) The second invokes `mgr-sync` to perform the refresh:

```
mgr-sync refresh --refresh-channels
```

Let me know if executing the commands separately works as expected.

**Comment by saltyollpheist** [comment]
Thanks for pointing out my error. I did correct it as you instructed. The following output is what occurred: 

`# spacewalk-common-channels \`
`> ubuntu-1804-pool-amd64-uyuni \`
`> ubuntu-1804-amd64-main-uyuni \`
`> ubuntu-1804-amd64-main-updates-uyuni \`
`> ubuntu-1804-amd64-main-security-uyuni \`
`> ubuntu-1804-amd64-universe-uyuni \`
`> ubuntu-1804-amd64-universe-updates-uyuni \`
`> ubuntu-1804-amd64-universe-security-uyuni \`
`> ubuntu-1804-amd64-uyuni-client`
`SUSE Manager username: <username>`
`SUSE Manager password: <password>`
No error occurred here.
`# mgr-sync refresh --refresh-channels`
`Refreshing Channel families     Please enter the credentials of SUSE Manager Administrator.`
`Login: <username>`
`Password: <password>`
 `              [DONE]`
`Refreshing SUSE products                       [FAIL]`
`        Error: <Fault -1: 'redstone.xmlrpc.XmlRpcFault: unhandled internal exception: No SCC organization credentials found.'>`

If this step is reliant upon an SCC organization credentials being setup, what is the correct command to refresh these channels I've just registered without the dependency on SCC? Thanks in advance.

**Comment by mackdk** [comment]
Indeed, you are right. This is kind of bug in the documentation. The command `mgr-sync refresh --refresh-channels` works only with vendor channels and SCC credentials. In your case you should use `spacewalk-repo-sync -p` followed by the name of the parent channel, to sync the parent and all its children. So, in your case:

```
spacewalk-repo-sync -p ubuntu-18.04-pool-amd64-uyuni
``` 

You can also use the command `spacecmd softwarechannel_syncrepos <CHANNEL ...>` to sync one or multiple channels.

I'll make sure we update the documentation and please let me know the synchronization works this time :slightly_smiling_face: .
